test/parallel_api/ranges/std_ranges_test.h

```diff
@@ -0,0 +1,450 @@
+    }
+};
+
+template<typename T, typename ViewType>
+struct usm_subrange_impl
+{
+    using shared_allocator = sycl::usm_allocator<T, sycl::usm::alloc::shared>;
+    using type = ViewType;
+
+    shared_allocator alloc;
+    T* p = NULL;
+    ViewType view;
+
+    template<typename Policy>
+    usm_subrange_impl(Policy&& exec, T* data, int n): alloc(exec.queue()), p(data)
+    {
+        auto mem = alloc.allocate(n);
+        view = ViewType(mem, mem + n);
+        std::copy_n(data, n, view.data());
+    }
+
+    ViewType& operator()()
+    {
+        return view;
+    }
+
+    ~usm_subrange_impl()
+    {
+        std::copy_n(view.data(), view.size(), p);
+        alloc.deallocate(view.data(), view.size());
+    }
+};
+
+template<typename T>
+using  usm_subrange = usm_subrange_impl<T, std::ranges::subrange<T*>>;
+
+template<typename T>
+using  usm_span = usm_subrange_impl<T, std::span<T>>;
+
+#endif // _ONEDPL_HETERO_BACKEND
+
+template<int call_id = 0, typename T = int, TestDataMode mode = data_in, bool RetTypeCheck = true, bool ForwardRangeCheck = false>
+struct test_range_algo
+{
+    void operator()(auto algo, auto checker, auto... args)
+    {
+
+        auto subrange_view = [](auto&& v) { return std::ranges::subrange(v); };
+        auto span_view = [](auto&& v) { return std::span(v); };
+
+        if constexpr(ForwardRangeCheck)
+        {
+            auto forward_view = [](auto&& v) {
+                using forward_it = TestUtils::ForwardIterator<decltype(v.begin()), ::std::forward_iterator_tag>;
+                return std::ranges::subrange(forward_it(v.begin()), forward_it(v.end()));
+            };
+            test<T, host_vector<T>, mode, RetTypeCheck>{}(host_policies(), algo, checker, forward_view, args...);
+        }
+
+        test<T, host_vector<T>, mode, RetTypeCheck>{}(host_policies(), algo, checker, subrange_view, std::identity{}, args...);
+        test<T, host_vector<T>, mode, RetTypeCheck>{}(host_policies(), algo, checker,  span_view, std::identity{}, args...);
+        test<T, host_vector<T>, mode, RetTypeCheck>{}(host_policies(), algo, checker, std::views::all, std::identity{}, args...);
+        test<T, host_subrange<T>, mode, RetTypeCheck>{}(host_policies(), algo, checker, std::views::all, std::identity{}, args...);
+        test<T, host_span<T>, mode, RetTypeCheck>{}(host_policies(), algo, checker, std::views::all, std::identity{}, args...);
+
+#if _ONEDPL_HETERO_BACKEND
+        //Skip the cases with pointer-to-function and hetero policy because pointer-to-function is not supported within kernel code.
+        if constexpr(!std::disjunction_v<std::is_member_function_pointer<decltype(args)>...>)
+        {
+            test<T, usm_vector<T>, mode, RetTypeCheck>{}(dpcpp_policy<call_id + 10>(), algo, checker, subrange_view, subrange_view, args...);
+            test<T, usm_vector<T>, mode, RetTypeCheck>{}(dpcpp_policy<call_id + 20>(), algo, checker, span_view, subrange_view, args...);
```

Comment:
The PR adds support of a std::vector being passed as a range. Should there be one more case without wrappers like `subrange_view` or `span_view`?

Reply:
For device policy `std::vector` is not supported, even with USM allocator.
For the host policies - I've added a test case with "pure" `std::vector`.

Reply:
> For device policy std::vector is not supported

What is the purpose of the function below?

```c++
#if _ONEDPL_CPP20_RANGES_PRESENT
    template <typename _T, sycl::usm::alloc _MemoryType>
    auto
    operator()(const std::vector<_T, sycl::usm_allocator<_T, _MemoryType>>& __vec) const
    {
#if _ONEDPL_CPP20_SPAN_PRESENT
        return std::span(__vec.begin(), __vec.end());
#else
        return std::ranges::subrange(__vec.begin(), __vec.end());
#endif
    }
#endif //_ONEDPL_CPP20_RANGES_PRESENT
```